db/db_impl.cc

```diff
@@ -599,7 +603,14 @@ void DBImpl::FindObsoleteFiles(JobContext* job_context, bool force,
     // find newly obsoleted log files
     while (alive_log_files_.begin()->number < min_log_number) {
       auto& earliest = *alive_log_files_.begin();
-      job_context->log_delete_files.push_back(earliest.number);
+      if (db_options_.recycle_log_files) {
+	Log(InfoLogLevel::INFO_LEVEL, db_options_.info_log,
+	    "adding log %" PRIu64 " to recycle list\n",
+	    earliest.number);
+	log_recycle_files.push_back(earliest.number);
```

Comment:
Do we recycle them infinitely, or can we cap the total size of files we can recycle? Just a suggestion. Not big deal if we don't cap it for now.